### ISSUE TITLE
`pipx`-installed `pre-commit` fails to create venvs as Python installation is not `healthy`

### ISSUE THREAD

**Issue by baggiponte** [pypa/pipx#818]
**Describe the bug**

TL;DR: when `pre-commit` is installed with `pipx`, it fails to create python venvs and returns the following error: `AssertionError: BUG: expected environment for python to be healthy() immediately after install, please open an issue describing your environment`. on the contrary, when i install `pre-commit` with homebrew, it works.

<details>
<summary> Full Traceback </summary>

```
Traceback (most recent call last):
  File "/Users/luca/.local/pipx/venvs/pre-commit/lib/python3.10/site-packages/pre_commit/error_handler.py", line 70, in error_handler
    yield
  File "/Users/luca/.local/pipx/venvs/pre-commit/lib/python3.10/site-packages/pre_commit/main.py", line 368, in main
    return hook_impl(
  File "/Users/luca/.local/pipx/venvs/pre-commit/lib/python3.10/site-packages/pre_commit/commands/hook_impl.py", line 237, in hook_impl
    return retv | run(config, store, ns)
  File "/Users/luca/.local/pipx/venvs/pre-commit/lib/python3.10/site-packages/pre_commit/commands/run.py", line 416, in run
    install_hook_envs(to_install, store)
  File "/Users/luca/.local/pipx/venvs/pre-commit/lib/python3.10/site-packages/pre_commit/repository.py", line 224, in install_hook_envs
    _hook_install(hook)
  File "/Users/luca/.local/pipx/venvs/pre-commit/lib/python3.10/site-packages/pre_commit/repository.py", line 86, in _hook_install
    raise AssertionError(
AssertionError: BUG: expected environment for python to be healthy() immediately after install, please open an issue describing your environment
```

</details>

**How to reproduce**

i installed `pipx` via homebrew (`brew install pipx`, as recommended by the official docs), and then used `pipx` to install `pre-commit`. what i get from `pipx`'s logs is that `pipx` itself uses homebrew python as default interpreter and not the one found on my `$PATH`. (by default, homebrew does not symlink python into `/usr/local` as there exist already the system python which, regrettably, is 2.7...):

<details>
  <summary>pipx pre-commit installation log</summary>

```
   619.2ms (setup:727): 2022-02-18 11:18:36
   619.3ms (setup:728): /usr/local/bin/pipx install pre-commit
   619.4ms (setup:729): pipx version is 1.0.0
   619.5ms (setup:730): Default python interpreter is '/usr/local/Cellar/pipx/1.0.0/libexec/bin/python3.10'
   620.1ms (package_name_from_spec:323): Determined package name: pre-commit
   620.1ms (package_name_from_spec:324): Package name determined in 0.0s
   621.1ms (run_subprocess:172): running /usr/local/Cellar/pipx/1.0.0/libexec/bin/python3.10 -m venv --without-pip /Users/luca/.local/pipx/venvs/pre-commit
   722.7ms (run_subprocess:185): stdout:
   723.0ms (run_subprocess:187): stderr:
   723.0ms (run_subprocess:188): returncode: 0
   723.8ms (run_subprocess:172): running /Users/luca/.local/pipx/venvs/pre-commit/bin/python -c import sysconfig; print(sysconfig.get_path('purelib'))
   781.1ms (run_subprocess:185): stdout: /Users/luca/.local/pipx/venvs/pre-commit/lib/python3.10/site-packages
   781.4ms (run_subprocess:188): returncode: 0
   782.1ms (run_subprocess:172): running /Users/luca/.local/pipx/shared/bin/python -c import sysconfig; print(sysconfig.get_path('purelib'))
   839.8ms (run_subprocess:185): stdout: /Users/luca/.local/pipx/shared/lib/python3.10/site-packages
   840.2ms (run_subprocess:188): returncode: 0
   841.0ms (run_subprocess:172): running /Users/luca/.local/pipx/venvs/pre-commit/bin/python --version
   855.9ms (run_subprocess:185): stdout: Python 3.10.2
   856.1ms (run_subprocess:187): stderr:
   856.2ms (run_subprocess:188): returncode: 0
   857.3ms (_parsed_package_to_package_or_url:128): cleaned package spec: pre-commit
   858.1ms (run_subprocess:172): running /Users/luca/.local/pipx/venvs/pre-commit/bin/python -m pip install pre-commit
  4694.6ms (run_subprocess:188): returncode: 0
  4696.8ms (run_subprocess:172): running <fetch_info_in_venv commands>
  4764.5ms (run_subprocess:185): stdout: {"sys_path": ["/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python310.zip", "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10", "/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/lib-dynload", "/Users/luca/.local/pipx/venvs/pre-commit/lib/python3.10/site-packages", "/Users/luca/.local/pipx/shared/lib/python3.10/site-packages"], "python_version": "3.10.2", "environment": {"implementation_name": "cpython", "implementation_version": "3.10.2", "os_name": "posix", "platform_machine": "x86_64", "platform_release": "21.2.0", "platform_system": "Darwin", "platform_version": "Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64", "python_full_version": "3.10.2", "platform_python_implementation": "CPython", "python_version": "3.10", "sys_platform": "darwin"}}
  4764.7ms (run_subprocess:188): returncode: 0
  4855.4ms (get_venv_metadata_for_package:303): get_venv_metadata_for_package: 159ms
  4855.8ms (_parsed_package_to_package_or_url:128): cleaned package spec: pre-commit
  4858.7ms (needs_upgrade:69): Time since last upgrade of shared libs, in seconds: 4969. Upgrade will be run by pipx if greater than 2592000.
  4869.1ms (cli:790): pipx finished.
```

</details>

when i add to the `.pre-commit-config.yaml` the hook for `isort`, the virtualenv is created under the python version in my `$PATH`. since the homebrew python is not on the path, i used`pyenv` to manage python versions on my machine and avoid the system python.

from what i understand, pyenv python is used to install `isort` in its separate env. this conflicts with the fact that `pipx` and `pre-commit` python interpreter is set to be homebrew python. yet, i don't understand if this problem lies within `pipx` or `pre-commit`. i am inclined to think it's the former, as i tried with a fresh `pre-commit` installed via homebrew and everything works. the funny thing is that this does not happen with `black`.

**Comment by Spitfire1900** [comment]
I did not reproduce this on Windows

**Comment by baggiponte** [comment]
I think this was a Homebrew - so macOS only - problem. I tried to go back and reproduce it myself and I could not: `pre-commit` installed with `pipx` works on my machine, so I will close the issue.
